### PR TITLE
Scale name first

### LINKED ISSUE #5535: First argument to scales is not longer the name of the scale
This seems to be a breaking change in the current development version. 

In ggplot2 version 3.4.4 (latest tag), ` scale_color_brewer("year")` would set `"year"` as the name of the scale:

``` r
library(ggplot2)
packageVersion("ggplot2")
#> [1] '3.4.4'
ggplot(mpg, aes(displ, hwy)) +
  geom_point(aes(colour = factor(year)), size = 5)  +
  scale_color_brewer("year")
```

In current dev version, this doesn't work:

``` r
library(ggplot2)
packageVersion("ggplot2")
#> [1] '3.4.4.9000'
ggplot(mpg, aes(displ, hwy)) +
  geom_point(aes(colour = factor(year)), size = 5)  +
  scale_color_brewer("year")
```

### PR DESCRIPTION
**Pull request by teunbrand**
This PR aims to fix #5535.

Briefly, it explicitly sets the first argument of many scales to the `name` argument. Thereby, we're circumventing the issue that `...` would funnel the first argument to the deprecated `scale_name` argument.

### CODE REVIEW

File: tests/testthat/test-scale-gradient.R [hunk 1]
```diff
@@ -12,11 +12,11 @@ test_that("points outside the limits are plotted as NA", {
 
 test_that("midpoints are transformed", {
 
-  scale <- scale_colour_gradient2(midpoint = 1, trans = "identity")
+  scale <- scale_colour_gradient2(midpoint = 1, transform = "identity")
   scale$train(c(0, 3))
   expect_equal(scale$rescale(c(0, 3)), c(0.25, 1))
 
-  scale <- scale_colour_gradient2(midpoint = 10, trans = "log10")
+  scale <- scale_colour_gradient2(midpoint = 10, transform = "log10")
```
Comment (teunbrand) [inline tests/testthat/test-scale-gradient.R:19]: This doesn't really belong to this PR per se, but I ran into this issue here so I thought to include it anyway.

**Review by thomasp85 — APPROVED** [review]
LGTM

**Comment by teunbrand** [comment]
Thanks for the review Thomas!
